### PR TITLE
Fix - Selecting all caption in Image block and pressing Backspace removes whole block

### PR DESCRIPTION
**Pull request by aspirisen**
Ticket: https://linear.app/prezly/issue/MT-4531/selecting-all-caption-in-image-block-and-pressing-backspace-removes

Demo:

### CODE REVIEW

File: packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx [hunk 1]
```diff
@@ -100,15 +99,6 @@ const ImageExtension = ({
...
             }
-
-            if (isDeletingBackward(event) && EditorCommands.isSelectionAtBlockStart(editor)) {
-                EditorCommands.removeNode(editor, {
-                    at: nodeEntry[1],
-                    match: isImageNode,
-                });
-                event.preventDefault();
-                event.stopPropagation();
-            }
```
Comment (e1himself) [inline packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx:111]: Does this mean it's not possible to delete the block by pressing backspace?

Scenario:

1. I have an image block
2. I have a text with paragraph after it
3. I put my cursor to the end of the paragraph
4. I press Backspace multiple types
5. I expect the paragraph contents to be deleted one character after another
6. Once the paragraph is gone I expect my cursor to:
   - Either remove the image block and make the document empty
   - Or move to the image caption text and start deleting it. 
     Once the caption is empty, my next Backspace pressing should remove the image block

How this scenario will work after this change?

Comment (aspirisen) [reply packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx:111]: @e1himself when the caption is empty and you press Backspace the image block will be removed and the selection will go to the block bellow

Comment (e1himself) [reply packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx:111]: Good! 

What about step 6 from the scenario above? Will it start deleting the caption or will it remove the image right away if I continue pressing Backspace after deleting the paragraph following the image block?

Comment (aspirisen) [reply packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx:111]: hm, i will take a look

Comment (e1himself) [reply packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx]: @aspirisen so what will be the behavior now?

Comment (aspirisen) [reply packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx]: @e1himself

Comment (e1himself) [reply packages/slate-editor/src/modules/editor-v4-image/ImageExtension.tsx]: For future reference: we'll follow up on improving the behavior with backspace and void elements in MT-4532 and MT-4533.

**Review by e1himself — APPROVED** [review]
Good job!

**Pull request merged**